openfisca_france/model/prelevements_obligatoires/impot_revenu/ir.py

```diff
@@ -1254,18 +1254,31 @@ def function(self, simulation, period):
 
 
 @reference_formula
-class decote(SimpleFormulaColumn):
+class decote(DatedFormulaColumn):
     column = FloatCol(default = 0)
     entity_class = FoyersFiscaux
     label = u"décote"
 
-    def function(self, simulation, period):
+    @dated_function(start = date(2001, 1, 1), stop = date(2013, 12, 31))
+    def function_2001_2013(self, simulation, period):
         period = period.start.offset('first-of', 'year').period('year')
         ir_plaf_qf = simulation.calculate('ir_plaf_qf', period)
         decote = simulation.legislation_at(period.start).ir.decote
 
         return period, (ir_plaf_qf < decote.seuil) * (decote.seuil - ir_plaf_qf) * 0.5
 
+    @dated_function(start = date(2014, 1, 1))
+    def function_2014(self, simulation, period):
```

Comment:
Thx, got it. Some IRPP date are given with a yearMonthDay notation, since fiscal rules apply usually only over a fiscal year, should't we change that to startYear_endYear notation ?